main.go

```diff
@@ -0,0 +1,178 @@
+package main
+
+import (
+	"crypto/tls"
+	"fmt"
+	"net"
+	"net/http"
+
+	"brank.as/serviceutil/logging"
+	"github.com/brankas/envcfg"
+	grpc_middleware "github.com/grpc-ecosystem/go-grpc-middleware"
+	grpc_logrus "github.com/grpc-ecosystem/go-grpc-middleware/logging/logrus"
+	grpc_recovery "github.com/grpc-ecosystem/go-grpc-middleware/recovery"
+	grpc_ctxtags "github.com/grpc-ecosystem/go-grpc-middleware/tags"
+	"github.com/grpc-ecosystem/grpc-gateway/runtime"
+	"github.com/improbable-eng/grpc-web/go/grpcweb"
+	"github.com/rs/cors"
+	"github.com/sirupsen/logrus"
+	"go.opencensus.io/plugin/ocgrpc"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/reflection"
+
+	"github.com/openbank/openbank/services"
+	"github.com/openbank/openbank/services/accounts"
+	"github.com/openbank/openbank/storage/inmemory"
+)
+
+var (
+	logger  *logrus.Entry
+	version = "devel"
+	service = "openbank"
+
+	config *envcfg.Envcfg
+)
+
+func init() {
+	logger = logging.NewLogger().WithFields(logrus.Fields{
+		"service": service,
+		"version": version,
+	})
+}
+
+func main() {
+	logger.Info("starting openbank service")
+
+	var err error
+	// Load the configuration
+	config, err = envcfg.New()
+	if err != nil {
+		logging.WithError(err, logger).Fatal("error loading configuration")
+	}
+
+	// TLS and grpc credentials
+	tlsConfig := config.TLS()
+	grpcCreds := credentials.NewTLS(tlsConfig)
+
+	// grpc
+	grpcServer, err := setupGRPCServer(grpcCreds)
+	if err != nil {
+		logging.WithError(err, logger).Fatal("unable to start grpc server")
+	}
+
+	// grpc-web handler
+	grpcWebServer := grpcweb.WrapServer(grpcServer)
+
+	// grpc-gateway
+	grpcGW, err := setupGRPCGateway()
+	if err != nil {
+		logging.WithError(err, logger).Fatal("failed to setup grpc gateway")
+	}
+
+	l, err := net.Listen("tcp", fmt.Sprintf(":%d", config.Port()))
+	if err != nil {
+		logging.WithError(err, logger).WithField("port", config.Port).Fatal("unable to listen on port")
+	}
+	logger.Infof("listening on %v", l.Addr())
+
+	// cors handler
+	c := cors.New(cors.Options{
+		AllowCredentials: true,
+		AllowedMethods: []string{
+			"HEAD", "GET", "POST",
+			"PUT", "PATCH", "DELETE",
+		},
+		AllowedOrigins: []string{
+			"*.brank.as",
```

Comment:
This should probably be updated? I assume we'll probably get a domain name to be used for all the openbank projects? I might be wrong too, and it might all be under `brank.as` domain.

Reply:
I don't know about that yet. We can still change that later :)

Reply:
yeah, definitely. This was just a thought when I was reading the code. =)